matchers.go

```diff
@@ -57,6 +57,11 @@ func HTTP1Fast(extMethods ...string) Matcher {
 	return PrefixMatcher(append(defaultHTTPMethods, extMethods...)...)
 }
 
+// TLS matches HTTPS requests.
+func TLS() Matcher {
```

Comment:
As far as the signature, how about something like:

>cmux.TLS(tls.VersionTLS10, tls.VersionTLS11)

And if no arguments are passed in should that just mean all versions?